File: tests/pages/base_page.py
from playwright.sync_api import Page, Locator


class BasePage:
    def __init__(self, page: Page):
        self.page = page

    def navigate_to(self, url: str):
        self.page.goto(url)

    def wait_for_element(self, locator: Locator, timeout: int = 10000):
        locator.wait_for(state="visible", timeout=timeout)

File: tests/pages/BasePage.ts
import { Page, Locator } from "@playwright/test";

export class BasePage {
  readonly page: Page;

  constructor(page: Page) {
    this.page = page;
  }

  async navigateTo(url: string) {
    await this.page.goto(url);
  }
  async waitForElement(locator: Locator, timeout: number = 10000) {
    await locator.waitFor({ state: "visible", timeout });
  }
}
